Components/Model/Sku.php

```diff
@@ -0,0 +1,172 @@
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @author Nosto Solutions Ltd <shopware@nosto.com>
+ * @copyright Copyright (c) 2016 Nosto Solutions Ltd (http://www.nosto.com)
+ * @license http://opensource.org/licenses/BSD-3-Clause BSD 3-Clause
+ */
+
+use Shopware_Plugins_Frontend_NostoTagging_Components_Helper_Price as PriceHelper;
+use Shopware\Models\Article\Detail as Detail;
+use Nosto\Object\Product\Sku as NostoSku;
+use Shopware\Models\Shop\Shop as Shop;
+use Nosto\Request\Http\HttpRequest as NostoHttpRequest;
+
+class Shopware_Plugins_Frontend_NostoTagging_Components_Model_Sku extends NostoSku
+{
+    protected static $ignoredCustomFieldsProperties = array(
+        'id',
+        'articleDetailId',
+        'articleDetail',
+        'articleId',
+        'article'
+    );
+
+    /**
+     * Loads the SKU Information
+     *
+     * @param Detail $detail Article Detail to load the SKU information
+     * @param Shop|null $shop the shop the product belongs to
+     */
+    public function loadData(Detail $detail, Shop $shop = null)
+    {
+        if (is_null($shop)) {
+            $shop = Shopware()->Shop();
+        }
+
+        $this->setUrl($this->assembleDetailUrl($detail, $shop));
+        $this->setId($detail->getId());
+        $this->setName($detail->getArticle()->getName());
+        $this->setImageUrl($this->getDetailImageUrl($detail));
+        $this->setPrice(floatval(PriceHelper::calcDetailPriceInclTax(
+            $detail,
+            $shop,
+            PriceHelper::PRICE_TYPE_NORMAL
+        )));
+        $this->setListPrice(floatval(PriceHelper::calcDetailPriceInclTax(
+            $detail,
+            $shop,
+            PriceHelper::PRICE_TYPE_LIST
+        )));
+        $this->setAvailable($this->isDetailAvailable($detail));
+        $this->setGtin($detail->getSupplierNumber());
+        $this->setCustomFields($this->getDetailCustomFields($detail));
+    }
+
+    /**
+     * Returns the URL for the given detail
+     * If no image is assigned, will return the parent
+     * article's main image
+     *
+     * @param Detail $detail
+     * @return null|string URL of detail image
+     */
+    protected function getDetailImageUrl(Detail $detail)
+    {
+        $mediaService = Shopware()->Container()->get('shopware_media.media_service');
```

Comment:
Should we move the logic to `Shopware_Plugins_Frontend_NostoTagging_Components_Helper_Image` to keep things consistent (we already have `getMainImageUrl` there)